### PR TITLE
internal/allocator: replace address traversal with ipaddr package

### PR DESCRIPTION
**Pull request by mdlayher**
**What this PR does / why we need it**:

Makes use of `ipaddr`'s address traversal functions, which can also deal with IPv6 addresses.  IPv4 addresses are still assumed in this PR because IPv6 addresses will need to be threaded through the code later.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes # n/a

**Special notes for your reviewer**:

n/a

### CODE REVIEW

File: internal/allocator/allocator.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"math"
 	"net"
 
+	"github.com/mikioh/ipaddr"
```
Comment (miekg) [inline internal/allocator/allocator.go]: nit, I think the order is: std lib, empty line, our libs, empty line, 3rd party

Comment (mdlayher) [reply internal/allocator/allocator.go]: Will do!

**Comment by mdlayher** [comment]
Note: I haven't actually tried this on my cluster as of yet.  I'm working on that but this is my first time doing the development setup.

**Comment by mdlayher** [comment]
Appears to be happy:
```
[zsh|matt@nerr-2]:~ 0 % k -n metallb-system logs controller-998f7675d-hx6k7
I0125 20:36:46.682827       1 main.go:134] MetalLB controller (commit d8d63f7-dirty, branch mdl-allocator-ipaddr)
W0125 20:36:46.777370       1 client_config.go:529] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0125 20:36:48.079825       1 main.go:107] Start config update
I0125 20:36:48.177379       1 main.go:120] End config update
I0125 20:36:48.177668       1 k8s.go:399] config changed, reconverging all services
I0125 20:36:48.178656       1 main.go:125] Controller synced, can allocate IPs now
I0125 20:37:17.863807       1 main.go:56] default/nginx: start update
I0125 20:37:17.863996       1 service.go:38] default/nginx: currently has no ingress IP
I0125 20:37:17.864069       1 service.go:70] default/nginx: needs an IP, allocating
I0125 20:37:17.864251       1 service.go:81] default/nginx: allocated "192.168.10.240"
I0125 20:37:17.864877       1 event.go:218] Event(v1.ObjectReference{Kind:"Service", Namespace:"default", Name:"nginx", UID:"88d7de15-020f-11e8-9d1c-b827eb0ccc6a", APIVersion:"v1", ResourceVersion:"918601", FieldPath:""}): type: 'Normal' reason: 'IPAllocated' Assigned IP "192.168.10.240"
I0125 20:37:18.381327       1 main.go:93] default/nginx: updated service status
I0125 20:37:18.381458       1 main.go:96] default/nginx: end update
I0125 20:37:18.381558       1 main.go:56] default/nginx: start update
I0125 20:37:18.382768       1 main.go:74] default/nginx: converged, no change
I0125 20:37:18.382888       1 main.go:75] default/nginx: end update
[zsh|matt@nerr-2]:~ 0 % k get service nginx 
NAME      TYPE           CLUSTER-IP     EXTERNAL-IP      PORT(S)        AGE
nginx     LoadBalancer   10.109.98.74   192.168.10.240   80:32518/TCP   33s
[zsh|matt@nerr-2]:~ 0 % curl -I http://192.168.10.240
HTTP/1.1 200 OK
Server: nginx/1.13.8
Date: Thu, 25 Jan 2018 20:38:57 GMT
Content-Type: text/html
Content-Length: 612
Last-Modified: Tue, 26 Dec 2017 11:11:22 GMT
Connection: keep-alive
ETag: "5a422e5a-264"
Accept-Ranges: bytes
```

**Pull request merged**